src/org/opendatakit/briefcase/ui/export/ExportForms.java

```diff
@@ -0,0 +1,102 @@
+package org.opendatakit.briefcase.ui.export;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.opendatakit.briefcase.model.BriefcaseFormDefinition;
+import org.opendatakit.briefcase.model.FormStatus;
+
+public class ExportForms {
+  private final List<FormStatus> forms;
+  private final Map<FormStatus, ExportConfiguration> configurations = new HashMap<>();
+  private Map<String, FormStatus> formsIndex = new HashMap<>();
+
+  ExportForms() {
+    this.forms = new ArrayList<>();
+  }
+
+  public ExportForms(List<FormStatus> forms) {
+    this.forms = forms;
+    refreshIndex();
+  }
+
+
+  public void merge(List<FormStatus> forms) {
+    this.forms.addAll(forms.stream().filter(form -> !this.formsIndex.containsKey(form.getFormDefinition().getFormId())).collect(toList()));
+    refreshIndex();
+  }
+
+  public int size() {
+    return forms.size();
+  }
+
+  public FormStatus get(int rowIndex) {
+    return forms.get(rowIndex);
+  }
+
+  public boolean hasConfiguration(FormStatus form) {
+    return configurations.containsKey(form);
+  }
+
+  public ExportConfiguration getConfiguration(FormStatus form) {
+    return configurations.computeIfAbsent(form, ___ -> ExportConfiguration.empty());
+  }
+
+  public Optional<ExportConfiguration> getConfiguration(BriefcaseFormDefinition formDefinition) {
+    return Optional.ofNullable(configurations.get(getForm(formDefinition)));
+  }
+
+  public void removeConfiguration(FormStatus form) {
+    configurations.remove(form);
+  }
+
+  public void setConfiguration(FormStatus form, ExportConfiguration configuration) {
+    configurations.put(form, configuration);
+  }
+
+  public void selectAll() {
+    forms.forEach(form -> form.setSelected(true));
+  }
```

Comment:
Don’t you miss Scala, where you could `_.setSelected` instead of `form -> form.setSelected`? Java’s making some steps in the right direction, though.

Reply:
So true!

In Java8 I like a more radical functional approach of currying a function: 

```java
Function2<Boolean,FormStatus,Void> setSelected = (value, form) -> form.setSelected(value)
```

and then using it on the `forEach` like so:

```java
forms.forEach(setSelected(true));
```